### PR TITLE
feat: add readonly modifier for antd table columns type

### PR DESCRIPTION
**Pull request by xiaoxiangmoe**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [x] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 💡 Background and solution

related issue: https://github.com/react-component/table/pull/574

support readonly array in antd table columns type

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Support ReadonlyArray in table columns |
| 🇨🇳 Chinese |  table columns 字段支持使用 ReadonlyArray    |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by tychenjiajun** [comment]
@zombieJ Can this be merged?

**Comment by tychenjiajun** [comment]
When can this be merged? The proposing change is exactly same as the source code in `rc-table`

**Comment by afc163** [comment]
1. CI broken
2. Please add typing test case

**Comment by xiaoxiangmoe** [comment]
ok, I' fix it tomorrow.

File: components/table/__tests__/type.test.tsx [hunk 1]
```diff
@@ -44,7 +44,7 @@ describe('Table.typescript types', () => {
       name: string;
     }
 
-    const columns: ColumnProps<User>[] = [
+    const columns: readonly ColumnProps<User>[] = [
```
Comment (afc163) [inline components/table/__tests__/type.test.tsx:47]: Please add a new type test case instead updating old one.

**Comment by li-jia-nan** [comment]
Hey! Conflicting files, and it seems this is now really outdated. If this is something that still makes sense to the latest master version of antd, please send a new PR with this change :)

**Pull request closed without merging**